build_tools/fish_xgettext.fish

```diff
@@ -22,34 +22,40 @@ set -l implicit_regex '(?:^| +)(?:complete|function).*? (?:-d|--description) (([
 # than messages which should be implicitly translated.
 set -l explicit_regex '.*\( *_ (([\'"]).+?(?<!\\\\)\\2) *\).*'
 
-rm -r /tmp/fish
+# Create temporary directory for these operations. OS X `mktemp` is somewhat restricted, so this block
+# works around that - based on share/functions/funced.fish.
+set -q TMPDIR
+or set -l TMPDIR /tmp
+set -l tmpdir (mktemp -d $TMPDIR/fish.XXXXXX)
+test "$tmpdir"
```

Comment:
I'd just directly do `or exit 1` here.

The $status is the one of `mktemp`, and if that fails it should return false. Otherwise your system is beyond repair.

Or at the very least use `test -n` or `test -z` - the one argument form is one of the worst mistakes in POSIX, and cause of most of our issues with `test`.

Reply:
> The $status is the one of mktemp, and if that fails it should return false. Otherwise your system is beyond repair.

Heh, fair enough :)